### DISCUSSION TITLE
BigInteger Usage

### DISCUSSION THREAD

**Discussion by kellyethridge** [kellyethridge/VBCorLib#100] [Questions]
This is a continuation of another discussion that should be its own. The other discussion can be found here https://github.com/kellyethridge/VBCorLib/discussions/92#discussion-43670

> ```
> Dim A As BigInteger 
> Set A = CBigInt(Text1.Text) 
> 
> Dim B As BigInteger 
> Set B = CBigInt(Text2.Text) 
> Set A = A.Modulus(B) 
> Text3.Text = A.ToString
> ```
> Is this the correct way to assign a value ?

The assignment of `BigInteger` seems appropriate. An alternate method might be
```
Text3.Text = BigInteger.Modulus(CBigInt(Text1.Text), CBigInt(Text2.Text)).ToString
```

I'm considering opening up the data types accepted in the `BigInteger` methods as described in https://github.com/kellyethridge/VBCorLib/issues/82#issue-723562110. This could simplify the previous example to
```
Text3.Text = BigInteger.Modulus(Text1.Text, Text2.Text).ToString
```

**Comment by kpsbahn** [comment]
Thank you very much.
The other method suggested by you, is crisp.
The proposed modification is even more succinct.
Best wishes,
